src/main/java/hu/bme/mit/spaceship/TorpedoStore.java

```diff
@@ -30,18 +30,18 @@ public TorpedoStore(int numberOfTorpedos){
 
   public boolean fire(int numberOfTorpedos){
     if(numberOfTorpedos < 1 || numberOfTorpedos > this.torpedoCount){
-      new IllegalArgumentException("numberOfTorpedos");
+      throw new IllegalArgumentException("numberOfTorpedos");
     }
 
     boolean success = false;
 
     // simulate random overheating of the launcher bay which prevents firing
-    Random generator = new Random();
+    Random generator =random;
```

Comment:
Legyen szóköz az egyenlőség két oldalán.

```suggestion
    Random generator = random;
```